lib/model.py

```diff
@@ -1,45 +1,73 @@
-from typing import List
+from typing import List, Tuple, Union
 
+import numpy as np
+import pandas as pd
 import tensorflow as tf
 
 
-@tf.keras.utils.register_keras_serializable()
-class OutputMapperLayer(tf.keras.layers.Layer):
-    """
-    The OutputMapperLayer converts the label indices produced by the model to
-    the taxonomy category ids and limits them to top N labels.
+@tf.function
+def top_labeled_predictions(
+        predictions: Union[tf.Tensor, np.array],
+        labels: List[str],
+        k: int = 10):
     """
+    Top labeled predictions.
+
+    This `@tf.function` can be used as a custom serving function.
+
+    Parameters
+    ----------
+    predictions: tf.Tensor or np.array
+        Predictions, as returned by `model.predict` or equivalent.
+
+    labels: List[str]
+        Label vocabulary.
 
-    def __init__(self, labels: List[str], top_n: int, **kwargs):
-        self.labels = labels
-        self.top_n = top_n
+    k: int, optional
+        Number of top predictions to return.
 
-        super(OutputMapperLayer, self).__init__(**kwargs)
+    Returns
+    -------
+    {'labels': tf.Tensor, 'scores': tf.Tensor}
```

Comment:
**IMPORTANT**: I changed the prediction output from a `(scores, labels)` tuple to a `{'labels': labels, 'scores': scores}` dict, in line with having more explicit data structures in the dataset.

It will change the model output in serving as well, so it'll require some changes in the code that uses the model for prediction. Is this OK, or should I revert to the tuple?